suite-native/module-send/src/sendFeesFormSchema.ts

```diff
@@ -1,11 +1,99 @@
 import { yup } from '@suite-common/validators';
+import { isDecimalsValid } from '@suite-common/wallet-utils';
+import { FeeInfo } from '@suite-common/wallet-types';
+import { NetworkType } from '@suite-common/wallet-config';
+import { BigNumber } from '@trezor/utils';
 
 import { NativeSupportedFeeLevel } from './types';
 
-const nativeSupportedFeeLevels: Array<NativeSupportedFeeLevel> = ['economy', 'normal', 'high'];
+type SendFeesFormContext = {
+    networkType?: NetworkType;
+    networkFeeInfo?: FeeInfo;
+    minimalFeeLimit?: string;
+};
+
+const nativeSupportedFeeLevels: Array<NativeSupportedFeeLevel> = [
+    'economy',
+    'normal',
+    'high',
+    'custom',
+];
 
 export const sendFeesFormValidationSchema = yup.object({
     feeLevel: yup.string().oneOf(nativeSupportedFeeLevels).required('Fee level is required'),
+    customFeePerUnit: yup
+        .string()
+        .required()
+        .test(
+            'too-many-decimals',
+            'Too many decimals.',
+            (value, { options: { context } }: yup.TestContext<SendFeesFormContext>) => {
+                if (!value) return true;
+
+                const { networkFeeInfo, networkType } = context!;
+                if (!networkFeeInfo || !networkType) return false;
+
+                if (networkType !== 'bitcoin' && networkType !== 'ethereum') return true;
+
+                let maxDecimals = 0;
+                if (networkType === 'bitcoin') {
+                    maxDecimals = 2;
+                } else if (networkType === 'ethereum') {
+                    maxDecimals = 9;
+                }
+
+                return isDecimalsValid(value, maxDecimals);
+            },
+        )
+        .test(
+            'fee-too-low',
+            'Fee is too low.',
+            (value, { options: { context } }: yup.TestContext<SendFeesFormContext>) => {
+                if (!value) return true;
+                const { networkFeeInfo } = context!;
+
+                if (!networkFeeInfo) return false;
+
+                const feeBig = new BigNumber(value);
```

Comment:
It is necessary to have BN here? How does numbers looks like?

I guess this validation runs with every change of form so it should be super quick.